### DISCUSSION TITLE
PacketTunnel exceeds 15M limit

### DISCUSSION THREAD

**Discussion by gaoxiaowei** [eycorsican/leaf#45] [Q&A]
我使用了ws over tls协议来测试，在访问www.yahoo.com时内存一直暴涨，会超过15M限制。请问这个有优化方法吗

**Comment by eycorsican** [comment]
I can't provide must suggestions on this kind of issues without any logs and configs.

I've discussed one of the reasons why the memory footprint would exceed 15MB in eycorsican/ileaf#26 , this specific issue could be avoided by using fakedns (always-fake-ip=*), and fakedns is safe to use with a recent fix, though the fix is not yet released, you can download a latest build at https://github.com/eycorsican/leaf/actions/runs/539108209 .

**Reply by gaoxiaowei** [reply]
I used the newly compiled libleaf.a with a memory footprint between 12-13M, but no crashes. It seems that the latest version has been optimized. Thank you very much
